examples/demo_magneticbottle.jl

```diff
@@ -51,51 +50,85 @@ stateinit = [r₀..., v₀...]
 param = prepare(getE, getB; species=Electron)
 tspan = (0.0, 3e-7)
 
-prob = ODEProblem(trace_relativistic!, stateinit, tspan, param)
+prob_rel = ODEProblem(trace_relativistic!, stateinit, tspan, param)
+prob_non = ODEProblem(trace!, stateinit, tspan, param)
 
-@printf "Speed = %6.4f %s\n" √(v₀[1]^2+v₀[2]^2+v₀[3]^2)/c*100 "% speed of light"
-@printf "Energy = %6.4f MeV\n" (1/√(1-(v₀[1]/c)^2-(v₀[2]/c)^2-(v₀[3]/c)^2)-1)*m*c^2/abs(q)/1e6
+vratio = √(v₀[1]^2+v₀[2]^2+v₀[3]^2)/c
+E = (1 / √(1 - vratio^2) - 1)*m*c^2/abs(q)/1e6
+
+v_str = @sprintf "Speed = %4.2f %s" vratio*100 "% speed of light"
+e_str = @sprintf "Energy = %6.4f MeV" E
+
+println(v_str)
+println(e_str)
 
 # Default Tsit5() and many solvers does not work in this case!
-sol = solve(prob, AB3(); dt=1.5e-12, save_idxs=[1,2,3])
+sol_rel = solve(prob_rel, AB3(); dt=1.5e-12, save_idxs=[1,2,3])
```

Comment:
Why do we need to use `AB3` here? I have encountered similar problems many times when calculating relativistic situations. Is there anything special about this algorithm, or did you discover it through trial and error?

Reply:
This was based on previous trials (on the wrong relativistic equations). In principle AB3 should be better for stiff problems, of which relativistic Lorentz equation may be. However, the applicability of non-stiff solvers like Tsit5 should be checked again together with other quantities like energy conservation.